### PR TITLE
Implement heatmap endpoints

### PR DESCRIPTION
**Pull request by nihar1024**
Active mobility endpoints implemented:
- Gravity based
- Closest Accessibility based
- Connectivity based

### CODE REVIEW

File: src/crud/crud_heatmap_connectivity.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    DefaultResultLayerName,
+)
+
+
+class CRUDHeatmapConnectivityBase(CRUDHeatmapBase):
+    TRAVELTIME_MATRIX_TABLE = {
+        ActiveRoutingHeatmapType.walking: "basic.traveltime_matrix_walking",
+        ActiveRoutingHeatmapType.bicycle: "basic.traveltime_matrix_bicycle",
+    }
+    H3_10_CELL_AREA_SQM = 11285.625000840355
```
Comment (EPajares) [inline src/crud/crud_heatmap_connectivity.py]: maybe you can compute this dynamically? There is a function for this.

File: src/crud/crud_heatmap_closest_average.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+class CRUDHeatmapClosestAverageBase(CRUDHeatmapBase):
+    TRAVELTIME_MATRIX_TABLE = {
+        ActiveRoutingHeatmapType.walking: "basic.traveltime_matrix_walking",
+        ActiveRoutingHeatmapType.bicycle: "basic.traveltime_matrix_bicycle",
+    }
+
+    def __init__(self, job_id, background_tasks, async_session, user_id, project_id):
+        super().__init__(job_id, background_tasks, async_session, user_id, project_id)
+
+    async def create_distributed_opportunity_table(self, layers: List[dict]):
```
Comment (EPajares) [inline src/crud/crud_heatmap_closest_average.py]: This functions is almost the same for CRUDHeatmapGravityBase

File: src/crud/crud_heatmap_connectivity.py (hunk 1 above)
Comment (nihar1024) [reply src/crud/crud_heatmap_connectivity.py]: For the connectivity heatmap query to be performant, I'm avoiding unnesting the array column but simply calculating array length * avg h3_10 area (after filtering by max_traveltime).

The library function h3_cell_area() takes a specific h3index as input, which would require unnesting the array, adding a lot of overhead to the query.

What I can do is put the avg area function `((3 * SQRT(3) / 2) * POWER(h3_get_hexagon_edge_length_avg(10, 'm'), 2))` into the query. I had hardcoded the value to further reduce overhead, but maybe this is fine.

**Comment by nihar1024** [comment]
Impedance functions to be confirmed:

Screenshot 2024-03-14 at 12 09 21 PM

**Pull request merged**